File: frontend/src/types/media.ts
/**
 * Типы для медиафайлов
 */

export type MediaType = 'image' | 'video' | 'audio' | 'voice' | 'file';

export interface MediaFile {
    id: string;
    user_id: string;
    filename: string;
    original_filename: string;
    media_type: MediaType;
    mime_type: string;
    size_bytes: number;
    url: string;
    created_at: string;
}

export interface UploadResponse {
    id: string;
    url: string;
    filename: string;
    media_type: MediaType;
    size_bytes: number;
}

export interface Attachment {
    id: string;
    media_file: MediaFile;
}

File: backend/src/schemas/media.py
"""Схемы для медиафайлов."""

import uuid
from datetime import datetime

from pydantic import BaseModel


class MediaFileResponse(BaseModel):
    """Ответ с данными медиафайла."""

    id: uuid.UUID
    media_type: str
    filename: str
    original_filename: str
    mime_type: str
    file_size: int
    url: str
    thumbnail_url: str | None
    duration_seconds: int | None
    width: int | None
    height: int | None
    created_at: datetime

    model_config = {"from_attributes": True}


class AttachmentResponse(BaseModel):
    """Ответ с данными вложения."""

    id: uuid.UUID
    media: MediaFileResponse
    order: int


class UploadResponse(BaseModel):
    """Ответ на загрузку файла."""

    media: MediaFileResponse
    message: str


class TranscriptionResponse(BaseModel):
    """Ответ с данными транскрипции."""

    id: uuid.UUID
    media_id: uuid.UUID
    status: str
    text: str | None
    language: str | None
    duration_seconds: float | None
    error_message: str | None
    created_at: datetime
    completed_at: datetime | None

    model_config = {"from_attributes": True}


class CreateTranscriptionRequest(BaseModel):
    """Запрос на создание транскрипции."""

    media_id: uuid.UUID
